### PR TITLE
IGNITE-20075 Get rid of potential thread leak in RocksDbTxStateStorageTest

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-20075

### CODE REVIEW

File: modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java [hunk 1]
```diff
@@ -43,18 +48,28 @@ public class RocksDbTxStateStorageTest extends AbstractTxStateStorageTest {
     @WorkDirectory
     private Path workDir;
 
+    private final ScheduledExecutorService scheduledExecutor = new ScheduledThreadPoolExecutor(1);
```
Comment (sashapolo) [inline modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java:51]: Not really an issue, but I would suggest to use `Executors.newSingleThreadScheduledExecutor`

File: modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java [hunk 2]
```diff
@@ -43,18 +48,28 @@ public class RocksDbTxStateStorageTest extends AbstractTxStateStorageTest {
...
+
+    private final ExecutorService executor = Executors.newFixedThreadPool(1);
```
Comment (sashapolo) [inline modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java:53]: Same here, we can change it to `Executors.newSingleThreadExecutor` for consistency sake

**Review by sashapolo — APPROVED** [review]

File: modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java [hunk 3]
```diff
@@ -43,18 +48,28 @@ public class RocksDbTxStateStorageTest extends AbstractTxStateStorageTest {
...
-                new ScheduledThreadPoolExecutor(1),
-                Executors.newFixedThreadPool(1),
+                scheduledExecutor,
+                executor,
                 () -> 1_000
         );
     }
 
+    @AfterEach
+    void shutdownExecutors() {
```
Comment (sashapolo) [inline modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java:68]: I'm a little bit worried that we stop the executors before we close the storage, maybe we should override the parent's `AfterEach` instead.

File: modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java (hunk 1 above)
Comment (tkalkirill) [reply modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java:51]: fix it

File: modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java (hunk 2 above)
Comment (tkalkirill) [reply modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java:53]: fix it

File: modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java (hunk 3 above)
Comment (tkalkirill) [reply modules/transactions/src/test/java/org/apache/ignite/internal/tx/storage/state/rocksdb/RocksDbTxStateStorageTest.java]: fix it

File: modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/storage/state/AbstractTxStateStorageTest.java [hunk 4]
```diff
@@ -75,8 +74,7 @@ void beforeTest() {
         tableStorage.start();
     }
 
-    @AfterEach
-    void afterTest() {
+    protected void afterTest() {
```
Comment (sashapolo) [inline modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/storage/state/AbstractTxStateStorageTest.java:77]: Why did you remove `@AfterEach` here?

Comment (tkalkirill) [reply modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/storage/state/AbstractTxStateStorageTest.java:77]: I tried to use the annotation `@Override` and `@AfterEach` at the same time and found that none of them is called, I read [here](https://stackoverflow.com/questions/48864716/before-beforeeach-inheritance-behaviour-change-junit4-junit5) that in JUnit5 you need to do it differently than in JUnit4, I did it.

Comment (tkalkirill) [reply modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/storage/state/AbstractTxStateStorageTest.java:79]: fix it

**Review by sashapolo — APPROVED** [review]
